spec/integration/inputs/kafka_spec.rb

```diff
@@ -91,6 +99,55 @@
         expect(event.get("[@metadata][kafka][timestamp]")).to be >= start
       end
     end
+
+    it "should show the right topic and group name in and kafka headers decorated kafka section" do
+      header = org.apache.kafka.common.header.internals.RecordHeader.new("name", "John".to_java_bytes)
+      record = org.apache.kafka.clients.producer.ProducerRecord.new(
+            "logstash_integration_topic_plain_with_headers", 0, "key", "value", [header])
+
+      send_message(record)
+      start = LogStash::Timestamp.now.time.to_i
+      consume_messages(decorate_headers_config, timeout: timeout_seconds, event_count: 1) do |queue, _|
+        expect(queue.length).to eq(1)
+        event = queue.shift
+        expect(event.get("[@metadata][kafka][topic]")).to eq("logstash_integration_topic_plain_with_headers")
+        expect(event.get("[@metadata][kafka][consumer_group]")).to eq(group_id_3)
+        expect(event.get("[@metadata][kafka][timestamp]")).to be >= start
+        expect(event.get("[@metadata][kafka][headers][name]")).to eq("John")
+      end
+    end
+
+    it "should skip headers not encoded in UTF-8" do
+      invalid = "😵".to_java_bytes
```

Comment:
Can you explain why this string is invalid? To me, it appears like you are encoding a UTF-8 emoji so the description of the spec leads me to believe it should work.

The below will report itself as UTF-8, but it contains an invalid byte sequence.

```suggestion
      invalid = "日本".encode(:Shift_JIS).force_encoding('UTF-8')
```

Reply:
I used the emoji since they are not valid UTF-8, but your suggest is far better